vmm/src/config.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use linux_loader::cmdline::Cmdline;
+use net_util::MacAddr;
+use std::convert::From;
+use std::net::Ipv4Addr;
+use std::path::Path;
+use std::result;
+use vm_memory::GuestAddress;
+
+pub const DEFAULT_VCPUS: &str = "1";
+pub const DEFAULT_MEMORY: &str = "512";
+pub const DEFAULT_RNG_SOURCE: &str = "/dev/urandom";
+const CMDLINE_OFFSET: GuestAddress = GuestAddress(0x20000);
+
+/// Errors associated with VM configuration parameters.
+#[derive(Debug)]
+pub enum Error {
+    /// Failed parsing cpus parameters.
+    ParseCpusParams(std::num::ParseIntError),
+    /// Failed parsing memory parameters.
+    ParseMemoryParams(std::num::ParseIntError),
+    /// Failed parsing kernel parameters.
+    ParseKernelParams,
+    /// Failed parsing kernel command line parameters.
+    ParseCmdlineParams,
+    /// Failed parsing disks parameters.
+    ParseDisksParams,
+    /// Failed parsing random number generator parameters.
+    ParseRngParams,
+    /// Failed parsing network parameters.
+    ParseNetParams,
+}
+pub type Result<T> = result::Result<T, Error>;
+
+pub struct VmParams<'a> {
+    pub cpus: &'a str,
+    pub memory: &'a str,
+    pub kernel: &'a str,
+    pub cmdline: Option<&'a str>,
+    pub disks: Vec<&'a str>,
+    pub rng: Option<&'a str>,
+    pub net: Option<&'a str>,
+}
+
+pub struct CpusConfig(pub u8);
+
+impl CpusConfig {
+    pub fn parse(cpus: &str) -> Result<Self> {
+        Ok(CpusConfig(
+            cpus.parse::<u8>().map_err(Error::ParseCpusParams)?,
+        ))
+    }
+}
+
+impl From<&CpusConfig> for u8 {
```

Comment:
You need some way to go the other way too.

Reply:
Not needed at the moment but something to consider.

Reply:
Yes we can think about this later.